server/jetstream_cluster.go

```diff
@@ -2735,6 +2736,14 @@ func (js *jetStream) monitorStream(mset *stream, sa *streamAssignment, sendSnaps
 							mset.retryMirrorConsumer()
 							continue
 						}
+						// If the error signals we timed out of a snapshot, we should try to replay the snapshot
+						// instead of fully resetting the state. Resetting the clustered state may result in
+						// race conditions and should only be used as a last effort attempt.
+						if errors.Is(err, errCatchupAbortedNoLeader) || err == errCatchupTooManyRetries {
+							if node := mset.raftNode(); node != nil && node.DrainAndReplaySnapshot() {
```

Comment:
What happens if DrainAndReplay returns false?

Reply:
Then we just return to the current behavior of calling reset clustered state.
However, since those errors only happen if there's a snapshot in the first place, this should never happen.